utils/common/src/main/java/brooklyn/util/javalang/AggregateClassLoader.java

```diff
@@ -69,8 +70,25 @@ public void add(int index, ClassLoader classLoader) {
         if (classLoader != null) classLoaders.add(index, classLoader);
     }
     
-    /** Returns the _live_ (and modifiable) list of classloaders 
-     * @return */ 
+    /** Resets the classloader shown here to be the given set */
+    public void reset(Collection<? extends ClassLoader> newClassLoaders) {
+        synchronized (classLoaders) {
```

Comment:
What is the point of the synchronized? Is it just to guard against multiple concurrent calls to `reset()`? None of the other modifications or accesses to `classLoaders` is synchronized.